From lcapitulino@mandriva.com.br Tue Jul 11 10:19:39 2006
Date: Tue, 11 Jul 2006 14:19:25 -0300
From: "Luiz Fernando N. Capitulino" <lcapitulino@mandriva.com.br>
To: Jon K Hellan <hellan@acm.org>
Cc: Greg KH <greg@kroah.com>
Subject: USB: Anydata: Fixes wrong URB callback.
Message-ID: <20060711141925.74c164cb@doriath.conectiva>


 Anydata is using usb_serial_generic_write_bulk_callback() for its
read URB, but it should use usb_serial_generic_read_bulk_callback()
instead (it's a read URB, isn't it?).

 Reported by Jon K Hellan <hellan@acm.org>.

Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/anydata.c |    2 +-
 drivers/usb/serial/generic.c |    1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/serial/anydata.c
+++ gregkh-2.6/drivers/usb/serial/anydata.c
@@ -71,7 +71,7 @@ static int anydata_open(struct usb_seria
 				  	  port->bulk_in_endpointAddress),
 			  port->read_urb->transfer_buffer,
 			  port->read_urb->transfer_buffer_length,
-			  usb_serial_generic_write_bulk_callback, port);
+			  usb_serial_generic_read_bulk_callback, port);
 	result = usb_submit_urb(port->read_urb, GFP_KERNEL);
 	if (result)
 		dev_err(&port->dev,
--- gregkh-2.6.orig/drivers/usb/serial/generic.c
+++ gregkh-2.6/drivers/usb/serial/generic.c
@@ -285,6 +285,7 @@ void usb_serial_generic_read_bulk_callba
 	if (result)
 		dev_err(&port->dev, "%s - failed resubmitting read urb, error %d\n", __FUNCTION__, result);
 }
+EXPORT_SYMBOL_GPL(usb_serial_generic_read_bulk_callback);
 
 void usb_serial_generic_write_bulk_callback (struct urb *urb, struct pt_regs *regs)
 {
